site/docs/advanced/flood.md

```diff
@@ -5,29 +5,119 @@ next: ./transformers.md
-In some cases it may make sense to use both.
+Deal with it.
+
+We have some good ideas about how many requests you can perform, but the exact numbers are unknown.
+(If someone tells you the limits, they are either not very well informed, or lying.)
+The limits are not simply hard thresholds that you can find out by experimenting with the Bot API.
+Rather, they are flexible constraints that change based on your bot's exact request payloads, the number of users, and other factors, not all of which are known.
+
+Here are a few misconceptions and false assumptions about rate limits.
+
+- My bot is too new to receive flood wait errors.
+- My bot does not get enough traffic to receive flood wait errors.
+- This feature of my bot is not used enough to receive flood wait errors.
+- My bot leaves enough time between API calls so it will not receive flood wait errors.
+- This particular method call cannot receive flood wait errors.
+- `getMe` cannot receive flood wait errors.
+- `getUpdates` cannot receive flood wait errors.
+
+All of these things are wrong.
+
+Let's get to the things we _do_ know.
+
+## Safe Assumptions About Rate Limits
+
+From the [Bot FAQ](https://core.telegram.org/bots/faq#my-bot-is-hitting-limits-how-do-i-avoid-this), we know a few limits that cannot be exceeded, ever.
+
+1. _"When sending messages inside a particular chat, avoid sending more than one message per second. We may allow short bursts that go over this limit, but eventually you'll begin receiving 429 errors."_
+
+   This one should be pretty clear. The auto-retry plugin will handle things for you.
+
+2. _"If you're sending bulk notifications to multiple users, the API will not allow more than 30 messages per second or so. Consider spreading out notifications over large intervals of 8—12 hours for best results."_
+
+   **This only applies to bulk notifications,** i.e. if you proactively message many users.
+   If you are just responding to messages from users, then it is no problem to send 1000 messages per second or more.
+
+   When the Bot FAQ says that you should _"consider spreading out notifications over large intervals"_ then this does not mean that you should add any artificial delays.
+   Instead, the main takeaway here is that sending bulk notifications is a process that will take many hours.
+   You cannot expect to message all users instantly at the same time.
+
+3. _"Also note that your bot will not be able to send more than 20 messages per minute to the same group."_
+
+   Again, pretty clear. Completely unrelated to bulk notifications or how many messages are sent in the group. And yet again, the auto-retry plugin will take care of this for you.
+
+## Throttling
+
+Some people think that it is bad to run into rate limits.
+They prefer to know the exact limits so they can throttle their bot.
+
+This is wrong.
+Rate limits are a useful tool for flood control, and if you respect them, this will have no negative consequences for your bot.
```

Comment:
```suggestion
Rate limits can be seen as a tool useful for flood control, and if you act accordingly, they won't have any negative impacts on your bot.
```

Reply:
“can be seen as” is incorrect for the TG Bot API because they are the singular tool to deal with this issue. This is not subjective.

I agree with the rest. How about this?

```suggestion
Rate limits are a tool useful for flood control, and if you act accordingly, they won't have any negative impacts on your bot.
```

Although “acting accordingly” sounds like we need to follow some rules in order to not insult someone. It feels a little off for “following API specifications.”

Reply:
Go with that.